File: backend/EmployeeManagement.Api/Models/Attendance.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeeManagement.Api.Models;

public static class AttendanceStatus
{
    public const string Present = "Present";
    public const string Absent = "Absent";
    public const string Late = "Late";
    public const string Excused = "Excused";
}

public class Attendance
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int EmployeeId { get; set; }

    [Required]
    public DateTime Date { get; set; }

    [Required]
    [MaxLength(20)]
    public string Status { get; set; } = AttendanceStatus.Present;

    public TimeSpan? CheckInTime { get; set; }

    public TimeSpan? CheckOutTime { get; set; }

    // Navigation property: Employee
    [ForeignKey(nameof(EmployeeId))]
    public Employee? Employee { get; set; }
}

File: frontend/src/types/attendance.ts
export type AttendanceStatus = 'Present' | 'Absent' | 'Late' | 'Excused';

export interface Attendance {
  id: number;
  employeeId: number;
  employeeName?: string;
  employeeEmail?: string;
  date: string;
  status: AttendanceStatus;
  checkInTime?: string;
  checkOutTime?: string;
}

export interface CreateAttendanceRequest {
  employeeId: number;
  date: string;
  status: AttendanceStatus;
  checkInTime?: string;
  checkOutTime?: string;
}

export interface UpdateAttendanceRequest {
  date: string;
  status: AttendanceStatus;
  checkInTime?: string;
  checkOutTime?: string;
}

export interface AttendanceQueryParams {
  employeeId?: number;
  date?: string;
  status?: string;
}
